projects/RabbitMQ.Client/client/api/ICredentialsRefresher.cs

```diff
@@ -82,12 +91,12 @@ public bool Unregister(ICredentialsProvider provider)
                 return false;
             }
 
-            var timer = _registrations[provider];
+            System.Timers.Timer timer = _registrations[provider];
```

Comment:
You might consider simplifying `Unregister` to the following:

```
public bool Unregister(ICredentialsProvider provider)
{
    if (!_registrations.TryRemove(provider, out var timer))
    {
        return false;
    }

    TimerBasedCredentialRefresherEventSource.Log.Unregistered(provider.Name);
    timer.Stop();
    timer.Dispose();
    return true;
}
```

Timer cannot be null, so no need for a null check.

Reply:
And no double lookup.

Reply:
Thanks for taking a look.